unified_planning/solvers/results.py

```diff
@@ -65,9 +71,34 @@ class PlanGenerationResult:
     def __post__init(self):
         # Checks that plan and status are consistent
         if self.status in POSITIVE_OUTCOMES and self.plan is None:
-            raise UPUsageError(f'The Result status is {self.status_as_str()} but no plan is set.')
+            raise UPUsageError(f'The Result status is {str(self.status)} but no plan is set.')
         elif self.status in NEGATIVE_OUTCOMES and self.plan is not None:
-            raise UPUsageError(f'The Result status is {self.status_as_str()} but the plan is {str(self.plan)}.\nWith this status the plan must be None.')
+            raise UPUsageError(f'The Result status is {str(self.status)} but the plan is {str(self.plan)}.\nWith this status the plan must be None.')
         self.metrics = {}
         self.log_messages = [] #NOTE Here, is this init right? Since it is done after the __init__ the value might be deleted
         return self
+
+@dataclass
+class ValidationResult:
+    '''Class that represents the result of a validate call.'''
+    status: ValidationResultStatus
+    error_info: str = ''
```

Comment:
Maybe we can optionally ask a validator to report the sequence of states? Something like `trace : List[State]`?
@alvalentini what do you think? We could re-use the state object we defined for simulated effects?

Reply:
We cannot reuse the same state object because the one we defined for simulated effects is an interface, but we can define a class that implement it.

However, I think that a `trace : List[State]` makes sense only for sequential plan validation, right?

Reply:
You are right, I'll open an issue and we postpone this.